verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSResourceRecordSetApi.java

```diff
@@ -0,0 +1,156 @@
+    public Iterator<ResourceRecordSet<?>> iterator() {
+
+        List<Record> recordList = api.getResourceRecordsList(domainName);
+        return VrsnContentConversionHelper.getSortedSetForDenominator(recordList).iterator();
+    }
+
+    public Iterator<ResourceRecordSet<?>> iterateByNameAndType(String name, String type) {
+        checkNotNull(type, "type was null");
+        checkNotNull(name, "name was null");
+
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        Iterator<ResourceRecordSet<?>> result = VrsnContentConversionHelper.getSortedSetForDenominator(recordList)
+                .iterator();
+        return result;
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterateByName(String name) {
+        try {
+            throw new VrsnMdnsException("Method Not Implemented", -1);
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Override
+    public ResourceRecordSet<?> getByNameAndType(String name, String type) {
+        checkNotNull(type, "type was null");
+        checkNotNull(name, "name was null");
+
+        ResourceRecordSet<?> result = null;
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        SortedSet<ResourceRecordSet<?>> tempSet = VrsnContentConversionHelper.getSortedSetForDenominator(recordList);
+        if (tempSet != null && tempSet.size() > 0) {
+            result = tempSet.first();
+        }
+        return result;
+    }
+
+    @Override
+    public void put(ResourceRecordSet<?> rrset) {
+
+        checkNotNull(rrset, "Resource Record was null");
+        checkNotNull(rrset.name(), "Resource Record Name was null");
+        checkNotNull(rrset.type(), "Resource Record Type was null");
+
+        // At this point disable delete
+        // as we might delete multiple records for name and type.
+        // Need to decide if that is correct
+        // ResourceRecordSet<?> rrsMatch = getByNameAndType(rrset.name(),
+        // rrset.type());
+        // if (rrsMatch != null) {
+        // deleteByNameAndType(rrset.name(), rrset.type());
+        // }
+        int ttlInt = 86000;
+        Integer ttlRRSet = rrset.ttl();
+        if (ttlRRSet != null) {
+            ttlInt = ttlRRSet.intValue();
+        }
+        String rData = getRDataStringFromRRSet(rrset);
+        api.createResourceRecord(domainName, rrset.type(), rrset.name(), "" + ttlInt, rData);
+
+    }
+
+    @Override
+    public void deleteByNameAndType(String name, String type) {
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        if (recordList != null && !recordList.isEmpty()) {
+            // delete all records in recordList
+            for (Record record : recordList) {
+                api.deleteRecourceRecord(domainName, record.id);
+            }
+
+        } else {
```

Comment:
behavior should be idempotent. in other words, if no records exist, then there's no reason to throw.